src/UbiHttp.cpp

```diff
@@ -0,0 +1,375 @@
+          "Connection: close\r\n"
+          "\r\n",
+          path, _host, _token, _user_agent);
+
+  if (_debug) {
+    Serial.println(F("Request sent"));
+    Serial.println(message);
+  }
+
+  _client_https_ubi.print(message);
+
+  while (_client_https_ubi.connected()) {
+    String line = _client_https_ubi.readStringUntil('\n');
+    if (line == "\r") {
+      if (_debug) {
+        Serial.println(F("Headers received"));
+      }
+      break;
+    }
+  }
+
+  free(message);
+  free(path);
+
+  double value = _parseServerAnswer();
+  _client_https_ubi.flush();
+  _client_https_ubi.stop();
+  // return 0;
+  return value;
+}
+
+double UbiHTTP::_parseServerAnswer() {
+
+  /**
+   * @param _charResponseLength 3 is the maximun amount of digits for the
+   * length to have
+   */
+  char *_charLength = (char *)malloc(sizeof(char) * 3);
+
+  readServerAnswer(_charLength);
+
+  /**
+   * The server respond the length of the value in HEX so it has to be
+   * converted to DEC
+   * */
+  uint8_t length = UbiUtils::hexadecimalToDecimal(_charLength);
+
+  if (_debug) {
+    Serial.print(F("Length: "));
+    Serial.println(length);
+  }
+
+  char *_charValue = (char *)malloc(sizeof(char) * length + 1);
+
+  readServerAnswer(_charValue);
+
+  double value = strtof(_charValue, NULL);
+
+  if (_debug) {
+    Serial.print("Value: ");
+    Serial.println(value);
+  }
+
+  free(_charLength);
+  free(_charValue);
+
+  return value;
+}
+
+/**
+ * @brief Calculate the lenght of the request line to be send over HTTP to the
+ * server
+ *
+ * @param path address of the endpoint to gather the data
+ * @return uint16_t  Lenght of the request line
+ */
+uint16_t UbiHTTP::_requestLineLength(char *path) {
+  uint16_t endpointLength = strlen("GET  HTTP/1.1\r\n"
+                                   "Host: \r\n"
+                                   "X-Auth-Token: \r\n"
+                                   "User-Agent: \r\n"
+                                   "Content-Type: application/json\r\n"
+                                   "Connection: close\r\n"
+                                   "\r\n") +
+                            strlen(path) + strlen(_host) + strlen(_token) +
+                            strlen(_user_agent);
+  return endpointLength;
+}
+
+/**
+ * @brief Calculate the lenght of the path to be send over HTTP to the server
+ *
+ * @param device_label device label of the device
+ * @param variable_label variable label to be updated or fetched
+ * @return uint16_t  Lenght of the endpoint path
+ */
+uint16_t UbiHTTP::_pathLength(const char *device_label,
+                              const char *variable_label) {
+  uint16_t endpointLength = strlen("/api/v1.6/devices///lv") +
+                            strlen(device_label) + strlen(variable_label);
+  return endpointLength;
+}
+
+/**
+ * @arg response [Mandatory] Pointer to store the server's answer
+ */
+
+void UbiHTTP::readServerAnswer(char *_serverResponse) {
+  bool _firstChar = true;
```

Comment:
refact this method, there are too many ifs that makes hard-reading the logic of the method